dags/dag_content/tower_metrics_content.py

```diff
@@ -0,0 +1,411 @@
+        ],
+        Tags=[],
+        EnableIAMDatabaseAuthentication=False,
+        DeletionProtection=False,
+        CopyTagsToSnapshot=False,
+    )
+    clone_db_info = {
+        "user": response["DBCluster"].get("MasterUsername"),
+        "host": response["DBCluster"].get("Endpoint"),
+        "resource_id": response["DBCluster"].get("DbClusterResourceId"),
+        "resource_arn": response["DBCluster"].get("DBClusterArn"),
+    }
+    # ensure cloning is complete before moving on
+    check_database_process_complete(
+        client=rds,
+        waiter_type='db_cluster_available',
+        db_name=clone_name
+    )
+
+    return clone_db_info
+
+@task
+def generate_random_password(aws_creds: dict) -> str:
+    """
+    generates random string password using boto3 secret client
+
+    Args:
+        aws_creds (dict): dictionary containing aws credentials
+
+    Returns:
+        str: generated password
+    """
+    secrets = create_secret_client(aws_creds)
+
+    response = secrets.get_random_password(
+        PasswordLength=30,
+        ExcludeCharacters="@",
+        ExcludePunctuation=True,
+        IncludeSpace=False,
+        RequireEachIncludedType=True,
+    )
+    password = response["RandomPassword"]
+    return password
+
+@task
+def modify_database_clone(aws_creds: dict, clone_name: str, password: str):
+    """
+    modifies cloned database cluster to have new master password
+
+    Args:
+        aws_creds (dict): dictionary containing aws credentials
+        clone_name (str): name of cloned database cluster
+        password (str): password generated in previous step
+    """
+    rds = create_rds_client(aws_creds)
+
+    rds.modify_db_cluster(
+        ApplyImmediately=True,
+        DBClusterIdentifier=clone_name,
+        MasterUserPassword=password,
+        EnableHttpEndpoint=True,
+    )
+    # ensure modification is complete before moving on
+    check_database_process_complete(client=rds,
+        waiter_type='db_cluster_available',
+        db_name=clone_name
+    )
+
+@task
+def update_secret(
+    aws_creds: dict, clone_name: str, db_info: dict, password: str
+) -> str:
+    """
+    updates secret in secret manager with formatted string includung new database info and random password
+
+    Args:
+        aws_creds (dict): dictionary containing aws credentials
+        clone_name (str): name of cloned database cluster
+        db_info (dict): dictionary with information from cloned database returned from clone_tower_cluster
+        password (str): randomly generated password assigned to cloned database
+
+    Returns:
+        str: secret arn string needed to access clone for queries
+    """
+    secrets = create_secret_client(aws_creds)
+
+    secret_string = (
+        "{"
+        + f'"dbInstanceIdentifier":"{clone_name}","engine":"aurora-mysql","host":"{db_info["host"]}","port":3306,"resourceId":"{db_info["resource_id"]}","username":"{db_info["user"]}","password":"{password}"'
+        + "}"
+    )
```

Comment:
You can use `json.dumps()` to serialize a Python dictionary to a string. 

```suggestion
    secret_dict = {
        "dbInstanceIdentifier": clone_name,
        "resourceId": db_info["resource_id"],
        "engine": "aurora-mysql",
        "host": db_info["host"],
        "port": 3306,
        "username": db_info["user"],
        "password": password,
    }
    secret_string = json.dumps(secret_dict)
```